storage/tianmu/core/query.cpp

```diff
@@ -1621,6 +1621,54 @@ int Query::BuildConditions(Item *conds, CondID &cond_id, CompiledQuery *cq, cons
   return RCBASE_QUERY_ROUTE;
 }
 
+bool Query::internalDataTimeIsNullToEq(THD *thd, Item *cond, Item **retcond) {

```

Comment:
i think that processing logic of datetime is beeter in `remove_eq_conds`.  and what the real value of `datetime` is stored in tianmu when we don't set the value and it allows null.

Reply:
At present, our tianmu engine only optimizes the conditions in the join object, but does not optimize the conditions in (select_lx). BuildConditions directly uses the conditions in (select_lx). At present, this modification method has the least impact on the system

Reply:
Use the condition in the join when the condition in the join is not null